test/parallel/test-fs-watchfile.js

```diff
@@ -19,8 +19,74 @@ assert.throws(function() {
   fs.watchFile(new Object(), function() {});
 }, /Path must be a string/);
 
-// Test ENOENT. Should fire once.
-const enoentFile = path.join(fixtures, 'empty', 'non-existent-file');
+const enoentFile = path.join(fixtures, 'non-existent-file');
+
+function customCloneObject(obj) {
+  const res = Object.create(null);
+  Object.keys(obj).forEach(function(key) {
+    res[key] = obj[key];
+  });
+  return res;
+}
+
+function areEmptyAndEqual(curr, prev) {
+  // create copies of arguments without the prototype information, because
+  // assert.deepStrictEqual compares the prototypes as well.
```

Comment:
@bnoordhuis It will use `undefined` for all the values.

    > new fs.Stats()
    { dev: undefined,
      mode: undefined,
      nlink: undefined,
      uid: undefined,
      gid: undefined,
      rdev: undefined,
      blksize: undefined,
      ino: undefined,
      size: undefined,
      blocks: undefined,
      atime: Invalid Date,
      mtime: Invalid Date,
      ctime: Invalid Date,
      birthtime: Invalid Date }

Otherwise, I should be doing something like this

    > const sDate = Date.UTC(1970, 0, 1, 0, 0, 0);
    undefined
    > new fs.Stats(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, sDate, sDate, sDate, sDate);
    { dev: 0,
      mode: 0,
      nlink: 0,
      uid: 0,
      gid: 0,
      rdev: 0,
      blksize: 0,
      ino: 0,
      size: 0,
      blocks: 0,
      atime: Thu Jan 01 1970 05:30:00 GMT+0530 (IST),
      mtime: Thu Jan 01 1970 05:30:00 GMT+0530 (IST),
      ctime: Thu Jan 01 1970 05:30:00 GMT+0530 (IST),
      birthtime: Thu Jan 01 1970 05:30:00 GMT+0530 (IST) }